### PR TITLE
clear all pixels when dispose is 3 but previousImage is null

### PR DESCRIPTION
**Pull request by michaeimm**
## Description
Fix Issues #2271 and part of #1094

## Motivation and Context
Dispose in frame should not be DISPOSAL_PREVIOUS when there did not save previous image. GIF like #2271 is wrong because the dispose is DISPOSAL_PREVIOUS in first frame.

### CODE REVIEW

**Comment by michaeimm** [comment]
I signed it!

**Comment by TWiStErRob** [comment]
Please run `gradlew check` locally, there seems to be some violation; probably a long line.

I also suggest you run this example with the new code to see if there's a regression:
https://github.com/TWiStErRob/glide-support/blob/master/src/glide3/java/com/bumptech/glide/supportapp/github/_1094_gif_disposal/TestFragment.java

File: third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java [hunk 1]
```diff
@@ -418,8 +418,9 @@ private Bitmap setPixels(GifFrame currentFrame, GifFrame previousFrame) {
     // Final location of blended pixels.
     final int[] dest = mainScratch;
 
-    // clear all pixels when meet first frame
-    if (previousFrame == null) {
+    // clear all pixels when meet first frame or dispose is 3 but previousImage is null
```
Comment (TWiStErRob) [inline third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java]: "dispose is 3 but previousImage is null" part doesn't explain much, it's exactly the same as the code below it, you should probably explain what it means, or why we need to drop the prev image/fill to black.

Comment (michaeimm) [reply third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java]: OK. I think conditions should be separate. We need to drop the prev image when meet first frame like we do clear all pixels. If prev image exist in second frame, prev image which saved in prev loop will be fill back in new loop and cause overdraw.

**Comment by sjudd** [comment]
:third_party:gif_decoder:checkstyle[ant:checkstyle] /home/travis/build/bumptech/glide/third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java:422: error: Line is longer than 100 characters (found 105).

https://travis-ci.org/bumptech/glide/jobs/293267074

**Comment by michaeimm** [comment]
Thank you for your reply.
I have modified something.
Please review the latest commit.

**Comment by sjudd** [comment]
The two travis failures are setup related and can be ignored.

File: third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java [hunk 2]
```diff
@@ -418,8 +418,17 @@ private Bitmap setPixels(GifFrame currentFrame, GifFrame previousFrame) {
     // Final location of blended pixels.
     final int[] dest = mainScratch;
 
-    // clear all pixels when meet first frame
+    // clear all pixels when meet first frame and drop prev image from last loop
     if (previousFrame == null) {
+      previousImage = null;
```
Comment (sjudd) [inline third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java]: Here you probably should be releasing previousImage so the Bitmap can be re-used, see the four lines in ``clear()``: https://github.com/bumptech/glide/blob/d447e582397cc7cf3f95a9871eb18d81a917544f/third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java#L327

**Comment by sjudd** [comment]
LGTM. Thank you for putting this up and your patience. I really appreciate the help!

**Pull request merged**